File: backend/sch/models/arrange_roster_request.py
from pydantic import BaseModel, Field


class OffRequest(BaseModel):
    worker_id: int = Field(alias="workerId")
    timeslots: list[str]


class ArrangeRosterRequest(BaseModel):
    team_id: int = Field(alias="teamId")
    timeslots: list[str]
    offs: list[OffRequest]

File: frontend/web/libs/server/roster/models/arrange/arrange-roster-request.ts
import { z } from "zod";
import { idSchema } from "../../../_general/models/id";

export const offRequestSchema = z.object({
  workerId: idSchema,
  days: z.date().array(),
})

export type OffRequest = z.infer<typeof offRequestSchema>;

export const arrangeRosterRequestSchema = z.object({
  departmentId: idSchema,
  days: z.date().array(),
  offs: offRequestSchema.array(),
})

export type ArrangeRosterRequest = z.infer<typeof arrangeRosterRequestSchema>;